pkg/engine/policycontext/policy_context.go

```diff
@@ -193,8 +193,14 @@ func NewPolicyContext(
 	configuration config.Configuration,
 ) (*PolicyContext, error) {
 	enginectx := enginectx.NewContext(jp)
-	if err := enginectx.AddResource(resource.Object); err != nil {
-		return nil, err
+	if operation != kyvernov1.Delete {
+		if err := enginectx.AddResource(resource.Object); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := enginectx.AddOldResource(resource.Object); err != nil {
+			return nil, err
+		}
```

Comment:
This change will be applied to all components that build policyContext using this method. The logic looks good to me, wonder how we handle the DELETE operation when fetching `request.oldObject` previously.

Per [doc](https://kyverno.io/docs/writing-policies/variables/#variables-from-admission-review-requests):
>{{request.object}}: the object being created or modified. It is null for DELETE requests.
{{request.oldObject}}: the object being modified. It is null for CREATE and CONNECT requests.

@chipzoller - for both in-cluster controllers and the CLI, can you confirm the user needs to specify `request.oldObject` to fetch object attributes if the operation is DELETE?

Reply:
Are you asking only if they write a policy which factors in resource contents when the operation is a DELETE?

Reply:
yes

Reply:
@eddycharly - can you help review this change?